src/wechaty/wechaty.py

```diff
@@ -363,24 +363,23 @@ async def ask_for_food(last_food, last_feed):
 
             self._watchdog.on('sleep', ask_for_food)
             self._watchdog.feed(food)
-            while True:
+            # If we set _watchdog to None or a new one, this loop should break.
+            while self._watchdog and id(self._watchdog) == watch_dog_id:
                 log.debug('bot tick <%s>', datetime.now())
                 await self._watchdog.sleep()
                 is_death = self._watchdog.starved_to_death()
                 if is_death:
                     await self.restart()
                     break
 
-        loop = asyncio.get_event_loop()
-        asyncio.run_coroutine_threadsafe(start_watchdog(), loop)
+        asyncio.create_task(start_watchdog())
```

Comment:
`create_task` will create a `task` for loop, but will not run util when you schedule it to loop. 

So, all of creating task and running coroutine are ok for running. 

```python
loop = asyncio.get_event_loop()
asyncio.run_coroutine_threadsafe(start_watchdog(), loop)
```

Reply:
Sorry, this is not the complete version, because this is about another issue, I was intended to store the watchdog task, and then cancel it when we call the `bot.stop()` action. But I transfer those change to another branch, I will roll back the changes on this issue first.

Reply:
Ok, waiting for your next working.

Reply:
I will start anther PR about that issue (perfect-restart) after this one has been merged.